plugin/src/main/java/org/autorefactor/refactoring/rules/SwitchRefactoring.java

```diff
@@ -385,4 +428,99 @@ private Variable extractVariableWithConstantValue(Expression firstOp, Expression
         }
         return null;
     }
+
+    @Override
+    public boolean visit(final SwitchStatement node) {
+        List<Pair<List<Statement>, List<Statement>>> switchStructure = getSwitchStructure(node);
+
+        for (int referenceIndex = 0; referenceIndex < switchStructure.size() - 1; referenceIndex++) {
+            for (int comparedIndex = referenceIndex + 1; comparedIndex < switchStructure.size(); comparedIndex++) {
+                Pair<List<Statement>, List<Statement>> referenceCase = switchStructure.get(referenceIndex);
+                Pair<List<Statement>, List<Statement>> comparedCase = switchStructure.get(comparedIndex);
+
+                if (!referenceCase.getSecond().isEmpty()
+                        && breaksControlFlow(referenceCase.getSecond().get(referenceCase.getSecond().size() - 1))
+                        && isTheSameCode(referenceCase.getSecond(), comparedCase.getSecond())) {
+                    List<Statement> precedingStatements = switchStructure.get(comparedIndex - 1).getSecond();
+
+                    if (breaksControlFlow(precedingStatements.get(precedingStatements.size() - 1))) {
+                        refactorByMergingCases(false, referenceCase, comparedCase);
+                        return DO_NOT_VISIT_SUBTREE;
```

Comment:
I don't understand. All our code have been coded this way. When a method always refactors, it returns nothing, when a method may refactor or not, we return a boolean. So in the first case, we then return `DO_NOT_VISIT_SUBTREE`.